### PR TITLE
Add npc interactions

### PR DESCRIPTION
**Pull request by echooff2**
#8

### CODE REVIEW

File: scenes/interactable.gd [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export(Shape2D) var shape = RectangleShape2D setget set_shape
+
+onready var _coll_shape_2d := $"CollisionShape2D" as CollisionShape2D
+
+
+func _ready():
+	assert(shape != null)
+	assert(_coll_shape_2d != null)
+
+	if shape != null:
```
Comment (AWarmCupOfTea) [inline scenes/interactable.gd:16]: This condition is always true because of the assertion above.

File: scenes/interactable.gd [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+		_coll_shape_2d.shape = shape
+
+
+func set_shape(new_shape):
+	shape = new_shape
+	if _coll_shape_2d != null:
```
Comment (AWarmCupOfTea) [inline scenes/interactable.gd:22]: Can this be false?

File: scenes/npc.gd [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+export(Shape2D) var shape = RectangleShape2D setget set_shape
+
+onready var _interactable := $"Interactable" as Interactable
+
+
+func _ready():
+	assert(shape != null)
+	assert(_interactable != null)
+
+	if shape != null:
```
Comment (AWarmCupOfTea) [inline scenes/npc.gd:15]: Same question as above

File: scenes/npc.gd [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+		_interactable.shape = shape
+
+
+func set_shape(new_shape: Shape2D):
+	shape = new_shape
+	if _interactable != null:
```
Comment (AWarmCupOfTea) [inline scenes/npc.gd:21]: Same question as above

File: scenes/player/interaction_box.gd [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+		# if distance is the same, first has prority
+		if distance < closest_distance:
+			closest = _interactables[i]
+			closest_distance = distance
+		
+	return closest
+
+
+func _on_InteractionBox_area_entered(area: Area2D):
```
Comment (AWarmCupOfTea) [inline scenes/player/interaction_box.gd:48]: Can we just check at the moment of action press, if player collides with any interactables? Not sure if we need to store them inside the player.

File: scenes/player/player.gd [hunk 6]
```diff
@@ -1,28 +1,33 @@
 class_name Player
 extends KinematicBody2D
 
+# should_player_stop: bool, some_dialog_info: int
```
Comment (AWarmCupOfTea) [inline scenes/player/player.gd:4]: Commented code.

File: scenes/player/player.gd [hunk 7]
```diff
@@ -34,5 +39,16 @@ func apply_gravity(delta: float, gravity_mult: float = 1.0) -> void:
...
 func can_jump() -> bool:
-	return self.is_on_floor()
+	return self.is_on_floor() and can_player_jump
+
+
+func can_interact() -> bool:
+	return can_player_interact
+
+
+func _on_Player_dialog_started(should_player_stop: bool, some_dialog_info: int):
```
Comment (AWarmCupOfTea) [inline scenes/player/player.gd:51]: This seems out of scope for this ticket: we don't even have any dialog system in place. I'd suggest removing this for now.

**Review by AWarmCupOfTea — CHANGES_REQUESTED** [review]

File: scenes/interactable.gd (hunk 2 above)
Comment (echooff2) [reply scenes/interactable.gd:22]: In editor it can be

File: scenes/npc.gd (hunk 4 above)
Comment (echooff2) [reply scenes/npc.gd:21]: In editor it can be

File: scenes/player/interaction_box.gd (hunk 5 above)
Comment (echooff2) [reply scenes/player/interaction_box.gd:48]: I did it this way because if we want to add in the future something like show text above npc on how to interact with it when it is the closest npc

File: scenes/player/player.gd (hunk 6 above)
Comment (echooff2) [reply scenes/player/player.gd:4]: It's not commented code. It's there just for documentation.

**Review by AWarmCupOfTea — APPROVED** [review]
